homeassistant/components/thermostat/ecobee.py

```diff
@@ -0,0 +1,263 @@
+thermostat has been added you can add the ecobee sensor component
+to your configuration.yaml.
+
+thermostat:
+  platform: ecobee
+  api_key: asdfasdfasdfasdfasdfaasdfasdfasdfasdf
+"""
+from homeassistant.loader import get_component
+from homeassistant.components.thermostat import (ThermostatDevice, STATE_COOL,
+                                                 STATE_IDLE, STATE_HEAT)
+from homeassistant.const import (
+    CONF_API_KEY, TEMP_FAHRENHEIT, STATE_ON, STATE_OFF)
+import logging
+import os
+
+REQUIREMENTS = [
+    'https://github.com/nkgilley/home-assistant-ecobee-api/archive/'
+    'e0388659a0f2fc7266485affbd398350cc0b5c58.zip#python-ecobee==0.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+ECOBEE_CONFIG_FILE = 'ecobee.conf'
+_CONFIGURING = {}
+
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """ Setup Platform """
+    # Only act if we are not already configuring this host
+    if 'ecobee' in _CONFIGURING:
+        return
+
+    setup_ecobee(hass, config, add_devices_callback)
+
+
+def setup_ecobee(hass, config, add_devices_callback):
+    """ Setup ecobee thermostat """
+    from pyecobee import Ecobee, config_from_file
+    # Create ecobee.conf if it doesn't exist
+    if not os.path.isfile(hass.config.path(ECOBEE_CONFIG_FILE)):
+        jsonconfig = {"API_KEY": config[CONF_API_KEY]}
+        config_from_file(hass.config.path(ECOBEE_CONFIG_FILE), jsonconfig)
+
+    ecobee = Ecobee(hass.config.path(ECOBEE_CONFIG_FILE))
+
+    # If ecobee has a PIN then it needs to be configured.
+    if ecobee.pin is not None:
+        # ecobee.request_pin()
+        request_configuration(ecobee, hass, add_devices_callback)
+        return
+
+    if 'ecobee' in _CONFIGURING:
+        _CONFIGURING.pop('ecobee')
+        configurator = get_component('configurator')
+        configurator.request_done('ecobee')
+
+    devices = []
+    for index in range(0, len(ecobee.thermostats)):
+        devices.append(Thermostat(ecobee, index))
+
+    add_devices_callback(devices)
+
+
+def request_configuration(ecobee, hass, add_devices_callback):
+    """ Request configuration steps from the user. """
+    configurator = get_component('configurator')
+    if 'ecobee' in _CONFIGURING:
+        configurator.notify_errors(
+            _CONFIGURING['ecobee'], "Failed to register, please try again.")
+
+        return
+
+    # pylint: disable=unused-argument
+    def ecobee_configuration_callback(data):
+        """ Actions to do when our configuration callback is called. """
+        ecobee.request_tokens()
+        ecobee.update()
+        setup_ecobee(hass, None, add_devices_callback)
+
+    _CONFIGURING['ecobee'] = configurator.request_config(
+        hass, "Ecobee", ecobee_configuration_callback,
+        description=(
+            'Please authorize this app at https://www.ecobee.com/consumer'
+            'portal/index.html with pin code: ' + ecobee.pin),
+        description_image='https://goo.gl/6tBkbH',
```

Comment:
Can you add this file to [homeassistant/components/frontend/www_static/images](https://github.com/balloob/home-assistant/tree/dev/homeassistant/components/frontend/www_static/images) ?

You can then link to it like this:
```python
description_image="/static/images/config_ecobee_thermostat.png"
```